### ISSUE TITLE
second argument to isinstance giving error for sequence of exceptons

### ISSUE THREAD

**Issue by BrandonNav** [microsoft/pyright#5018]
**Describe the bug**
I just upgrade to 1.1.305 and observed that isinstance now fails when checking instances of exceptions

**Error Message**
```bash
test.py:7:34 - error: Second argument to "isinstance" must be a class or tuple of classes
    TypeVar or generic type with type arguments not allowed
```

**To Reproduce run pyright on the following code**
```python
from collections.abc import Sequence


exceptions: Sequence[type[BaseException]] = (KeyboardInterrupt, SystemExit, EOFError)

def tester(exception: Exception) -> bool:
    return isinstance(exception, exceptions). # error: Second argument to "isinstance" must be a class or tuple of ...
```

**VS Code extension or command-line**
Pyright: 1.1.305 on the cli

Pylance (v2023.4.40) does not report an error

**Comment by erictraut** [comment]
I'm not able to repro the problem. Are there any other repro steps? Any non-default configuration options?

Can you verify that you can repro with a clean venv, a new project, and the code copied directly from the bug report above?

I don't think it matters in this case, but which version of Python are you using?

**Comment by BrandonNav** [comment]
@erictraut apologies, I should have taken those steps too.

I am using Python 3.11.1.
My pyproject.toml config is:
```toml
[tool.pyright]
pythonVersion = "3.11"
pythonPlatform = "All"
typeCheckingMode = "basic"
useLibraryCodeForTypes = false
reportMissingImports = true
reportUnnecessaryTypeIgnoreComment = true

include = [
    "*.py",
]
exclude = [
    "**/__pycache__",
    "**/local_modules",
    "**/migrations",
    "**/tests/",
    "**/tests.py",
]
```

To retest, I deleted the virtualenv and reran pyright with my configuration and still got the issue.

I fear my minimal reproduction code is too minimal, despite it occurring for me.
Here is the code again but similar to how my codebase is structured

```python
from __future__ import annotations

from collections.abc import Callable, Sequence


def wrapper(exceptions: Sequence[type[BaseException]]) -> Callable[[Exception], bool]:
    def tester(exception: Exception) -> bool:
        return isinstance(exception, tuple(exceptions))

    return tester
```
<img width="1017" alt="image" src="https://user-images.githubusercontent.com/30785148/234742870-800b6c93-2535-4fc1-b4c4-68f63a1764eb.png">

Are there additional diagnostics I could run?

**Comment by erictraut** [comment]
Thanks, I'm now able to repro the problem. I have a fix that will be included in the next release. 

If you're interested, the problem is related to the fact that the typeshed type definition for the `isinstance` and `issubclass` functions uses a type annotation for the second parameter that is a recursive type alias. 

```python
_ClassInfo: TypeAlias = type | types.UnionType | tuple[_ClassInfo, ...]
```

Your use case involved the recursive portion of the type alias (the `tuple[_ClassInfo, ...]`), and pyright's logic for checking whether the type was compatible with the additional requirements of `isinstance` was tripping up on this.

**Comment by BrandonNav** [comment]
Ah that makes sense. Thank you for the prompt fix

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.306, which I just published. It will also be included in a future release of pylance.
